tools/bootstrap_commitlogs/main/main.go

```diff
@@ -0,0 +1,233 @@
+	if strings.TrimSpace(shards) != "" {
+		for _, shard := range strings.Split(shards, ",") {
+			shard = strings.TrimSpace(shard)
+			if shard == "" {
+				log.Fatalf("Invalid shard list: '%s'", shards)
+			}
+			value, err := strconv.Atoi(shard)
+			if err != nil {
+				log.Fatalf("could not parse shard '%s': %v", shard, err)
+			}
+			rng := xtime.Range{Start: startInclusive, End: endExclusive}
+			shardTimeRanges[uint32(value)] = xtime.NewRanges().AddRange(rng)
+			shardsAll = append(shardsAll, uint32(value))
+		}
+		// Or just handled up to N (shard-count) shards
+	} else if shardsCount > 0 {
+		for i := uint32(0); i < uint32(shardsCount); i++ {
+			rng := xtime.Range{Start: startInclusive, End: endExclusive}
+			shardTimeRanges[i] = xtime.NewRanges().AddRange(rng)
+			shardsAll = append(shardsAll, i)
+		}
+	} else {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	log.WithFields(
+		xlog.NewField("pathPrefix", pathPrefix),
+		xlog.NewField("namespace", namespaceStr),
+		xlog.NewField("shards", shardsAll),
+	).Infof("configured")
+
+	instrumentOpts := instrument.NewOptions().
+		SetLogger(log)
+
+	retentionOpts := retention.NewOptions().
+		SetBlockSize(blockSize).
+		SetRetentionPeriod(bootstrapRetention).
+		SetBufferPast(1 * time.Minute).
+		SetBufferFuture(1 * time.Minute)
+
+	blockOpts := block.NewOptions()
+
+	encoderPool := encoding.NewEncoderPool(nil)
+	iteratorPool := encoding.NewReaderIteratorPool(nil)
+	multiIteratorPool := encoding.NewMultiReaderIteratorPool(nil)
+	segmentReaderPool := xio.NewSegmentReaderPool(nil)
+
+	encodingOpts := encoding.NewOptions().
+		SetEncoderPool(encoderPool).
+		SetReaderIteratorPool(iteratorPool).
+		SetBytesPool(blockOpts.BytesPool()).
+		SetSegmentReaderPool(segmentReaderPool)
+
+	encoderPool.Init(func() encoding.Encoder {
+		return m3tsz.NewEncoder(time.Time{}, nil, true, encodingOpts)
+	})
+
+	iteratorPool.Init(func(r io.Reader) encoding.ReaderIterator {
+		return m3tsz.NewReaderIterator(r, true, encodingOpts)
+	})
+
+	multiIteratorPool.Init(func(r io.Reader) encoding.ReaderIterator {
+		iter := iteratorPool.Get()
+		iter.Reset(r)
+		return iter
+	})
+
+	segmentReaderPool.Init()
+
+	blockPool := block.NewDatabaseBlockPool(nil)
+	blockPool.Init(func() block.DatabaseBlock {
+		return block.NewDatabaseBlock(time.Time{}, ts.Segment{}, blockOpts)
+	})
+
+	blockOpts = blockOpts.
+		SetEncoderPool(encoderPool).
+		SetReaderIteratorPool(iteratorPool).
+		SetMultiReaderIteratorPool(multiIteratorPool).
+		SetDatabaseBlockPool(blockPool).
+		SetSegmentReaderPool(segmentReaderPool)
+
+	resultOpts := result.NewOptions().
+		SetInstrumentOptions(instrumentOpts).
+		SetDatabaseBlockOptions(blockOpts)
+
+	fsOpts := fs.NewOptions().
+		SetInstrumentOptions(instrumentOpts).
+		SetFilePathPrefix(pathPrefix)
+
+	commitLogOpts := commitlog.NewOptions().
+		SetInstrumentOptions(instrumentOpts).
+		SetFilesystemOptions(fsOpts).
+		SetFlushSize(flushSize).
+		SetBlockSize(blockSize).
+		SetReadConcurrency(8)
```

Comment:
maybe take a flag for this too

Reply:
I actually have to take these out for now because that code is not merged yet